lib/FileUploader/utils/tests/uploadFile-test.js

```diff
@@ -0,0 +1,119 @@
+import {
+  describe,
+  it,
+  beforeEach,
+  afterEach,
+} from '@bigtest/mocha';
+import { expect } from 'chai';
+import Pretender from 'pretender';
+
+import { uploadFile } from '../uploadFile';
+
+function createUploadResponse(server, response, status = 200) {
+  server.post('/uploadUrl', () => [status, { 'content-type': 'application/json' }, JSON.stringify(response)]);
+}
+
+describe('uploadFile', () => {
+  let pretender;
+
+  beforeEach(() => {
+    pretender = new Pretender();
+  });
+
+  afterEach(() => {
+    pretender.shutdown();
+  });
+
+  describe('getting successful response upon upload', () => {
+    let response;
+    const responsePayload = { message: 'success' };
+
+    beforeEach(() => {
+      createUploadResponse(pretender, responsePayload);
+    });
+
+    beforeEach(async () => {
+      response = await uploadFile({
+        xhr: new XMLHttpRequest(),
+        url: '/uploadUrl',
+        file: new File([], 'file.js'),
+        okapi: {},
+      });
+    });
+
+    it('should be handled correctly', () => {
+      expect(response.message).to.equal(responsePayload.message);
+    });
+  });
+
+  describe('getting API error upon file upload', () => {
+    let response;
+    const responsePayload = { message: 'error' };
+
+    beforeEach(() => {
+      createUploadResponse(pretender, responsePayload, 500);
+    });
+
+    beforeEach(async () => {
+      try {
+        response = await uploadFile({
```

Comment:
It seems, the same beforeEach callback is used in several places. Are there any ways to move it one function/handler to prevent duplication? Or do we expect that these handlers will be changes in the future?

Reply:
Good idea. Will create reusable function for that.